lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -23,6 +26,43 @@ def with(n)
         end
 
         # @api public
+        # Specifies keyword arguments, if any.
+        #
+        # @example
+        #   expect(obj).to respond_to(:message).with_keywords(:color, :shape)
+        # @example with an expected number of arguments
+        #   expect(obj).to respond_to(:message).with(3).arguments.and_keywords(:color, :shape)
+        def with_keywords(*keywords)
+          @expected_keywords = keywords
+          self
+        end
+        alias :and_keywords :with_keywords
```

Comment:
Doesn't hurt anything, I think, and makes the interface more fluent. Just to play devil's advocate, there are a couple of alternatives I can think of:

- Have the 'and_' methods raise an error if there is not already an expectation. Relatively simple to implement and test, but adds a bit of complexity and possibly frustration (e.g. if a user modifies an existing test from `respond_to().with().and_...` to `respond_to().and_...).

- Have the 'with` methods return an object that wraps the matcher and adds the `and_` methods. Much more complicated to implement and test.

Ultimately at least in my opinion the simplicity and being consistent with the object API outweighs the "I can do something kinda funny sounding but it works". As a comparison, RSpec doesn't require composable aliases - you can say `expect().to include(be_a String)`, even if the message would make more sense if you used the `an_instance_of` alias.